### PR TITLE
Remove debug symbols from non-symbols ilc package

### LINKED ISSUE #102684: .dbg in ilcompiler package
They can be in a separate symbols package:

```sh
$ find ~/.nuget/packages/runtime.linux-arm64.microsoft.dotnet.ilcompiler -type f -name '*dbg' -exec du -sh {} \; | sort -rh

24M	/root/.nuget/packages/runtime.linux-arm64.microsoft.dotnet.ilcompiler/9.0.0-preview.4.24266.19/tools/ilc.dbg
1.2M	/root/.nuget/packages/runtime.linux-arm64.microsoft.dotnet.ilcompiler/9.0.0-preview.4.24266.19/framework/libSystem.IO.Compression.Native.so.dbg
504K	/root/.nuget/packages/runtime.linux-arm64.microsoft.dotnet.ilcompiler/9.0.0-preview.4.24266.19/framework/libSystem.Security.Cryptography.Native.OpenSsl.so.dbg
336K	/root/.nuget/packages/runtime.linux-arm64.microsoft.dotnet.ilcompiler/9.0.0-preview.4.24266.19/framework/libSystem.Native.so.dbg
228K	/root/.nuget/packages/runtime.linux-arm64.microsoft.dotnet.ilcompiler/9.0.0-preview.4.24266.19/framework/libSystem.Globalization.Native.so.dbg
40K	/root/.nuget/packages/runtime.linux-arm64.microsoft.dotnet.ilcompiler/9.0.0-preview.4.24266.19/framework/libSystem.Net.Security.Native.so.dbg
```

ilc.dbg is the second largest file in this package.

### PR DESCRIPTION
**Pull request by am11**
The packs subsets creates two packages:

`Microsoft.DotNet.ILCompiler.9.0.0-<VersionSuffix>.nupkg`
`Microsoft.DotNet.ILCompiler.9.0.0-<VersionSuffix>.symbols.nupkg`

The non-symbols package also contain debug symbols files which increases the size of the package.

This PR uses the existing, unused, property `IncludeSymbolsInPackage` to filter out symbol files from non-symbols package.

Fixes #102684

### CODE REVIEW

File: src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <GeneratePackage>true</GeneratePackage>
-    <IncludeSymbolsInPackage>true</IncludeSymbolsInPackage>
```
Comment (MichalStrehovsky) [inline src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:5]: Does this remove things like Runtime.ServerGC.pdb (on Windows) or System.Private.CoreLib.pdb (on all platforms)?

We deliberately want them in the package.

Comment (jkotas) [reply src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:5]: Yes, we want .pdbs for runtime binaries in the non-symbol package.

If you remove the .pdbs for native object files from the package, we will end up with a ton of warnings during `dotnet publish`:
```
  Generating native code
bootstrapper.obj : warning LNK4099: PDB '' was not found with 'bootstrapper.obj' or at ''; linking object as if no debug
info [C:\repro\repro.csproj]
Runtime.WorkstationGC.lib(threadstore.cpp.obj) : warning LNK4099: PDB '' was not found with 'Runtime.WorkstationGC.lib(th
readstore.cpp.obj)' or at ''; linking object as if no debug info [C:\repro\repro.csproj]
Runtime.WorkstationGC.lib(EHHelpers.cpp.obj) : warning LNK4099: PDB '' was not found with 'Runtime.WorkstationGC.lib(EHHe
lpers.cpp.obj)' or at ''; linking object as if no debug info [C:\repro\repro.csproj]
....
```
The .pdbs for managed runtime binaries are not needed at the moment, but they will be needed for #68714

Comment (am11) [reply src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:5]: Thanks. I will try to figure out how to only filter out .dbg from non-symbols package (keeping managed pdbs intact). FWIW, in `runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.5.24272.2` package, total size of .pdb files is  9.2 MB.

Comment (am11) [reply src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:5]: Changed such that `Microsoft.DotNet.ILCompiler.<Version>.nupkg` contains pdb and `.symbols.nupkg` contains  both pdb and dbg/dSYM/debug etc.

File: src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
+    <IncludeSymbolsInPackage>false</IncludeSymbolsInPackage>
```
Comment (jkotas) [inline src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:5]:
```suggestion
```
Is this needed? false should be the default.

File: src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj [hunk 3]
```diff
@@ -28,6 +29,14 @@
       <File Include="$(CoreCLRBuildIntegrationDir)*" TargetPath="build" />
       <File Include="$(CoreCLRILCompilerDir)netstandard\*" TargetPath="tools/netstandard" />
     </ItemGroup>
+
+    <!-- exclude unix-y native symbols from ilc package (they are included in symbols package) -->
```
Comment (jkotas) [inline src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:33]: Does this mean that we are still including `ilc.pdb` in the ilc package on Windows?

Comment (am11) [reply src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:32]: Yup, I modeled it after @MichalStrehovsky's comment:

> Does this remove things like Runtime.ServerGC.pdb (on Windows) or System.Private.CoreLib.pdb (on all platforms)?
>
> We deliberately want them in the package.

If we want to apply it for Windows as well, we have a way to distinguish native vs. managed pdb in nearby context.

**Review by agocke — APPROVED** [review]
LGTM. Unfortunate about the Windows situation. Maybe if we get the ability to restore symbols at publish time that will remove the need to put the managed PDBs in the main package.

File: src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj (hunk 3 above)
Comment (jkotas) [reply src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:32]: The distintion should be aot runtime vs. tools. The symbols should be stripped for tools (both managed and native), but kelp for runtime bits.

The tools are under `tools` directory in the package. Can we do the filter based on the directory?

File: src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj [hunk 4]
```diff
@@ -28,6 +29,14 @@
...
+    <ItemGroup>
+      <LibPackageExcludes Include="%2A%2A\%2A.dbg"/>
```
[the remark this answers is not in the document]
Comment (jkotas) [reply src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj]: I still see ilc.pdb in runtime.win-x64.Microsoft.DotNet.ILCompiler.9.0.0-dev.nupkg with the latest changes.

[the remark this answers is not in the document]
Comment (am11) [reply src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj]: Ah, the prefix was incorrect. Should had tested on Windows machine. 🤦

File: src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj [hunk 5]
```diff
@@ -28,6 +28,17 @@
       <File Include="$(CoreCLRBuildIntegrationDir)*" TargetPath="build" />
       <File Include="$(CoreCLRILCompilerDir)netstandard\*" TargetPath="tools/netstandard" />
     </ItemGroup>
+
+    <!-- exclude native symbols from ilc package (they are included in symbols package) -->
+    <ItemGroup>
+       <!-- on windows, remove the pdbs only from tools directory (both managed and native) -->
+      <LibPackageExcludes Include="$(CoreCLRILCompilerDir)%2A%2A\%2A.pdb"/>
```
Comment (am11) [inline src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:35]:
```suggestion
      <LibPackageExcludes Include="tools\%2A%2A\%2A.pdb"/>
```

File: src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj [hunk 6]
```diff
@@ -28,6 +28,17 @@
...
     </ItemGroup>
+
+    <!-- exclude native symbols from ilc package (they are included in symbols package) -->
+    <ItemGroup>
+       <!-- on windows, remove the pdbs only from tools directory (both managed and native) -->
+      <LibPackageExcludes Include="tools\%2A%2A\%2A.pdb"/>
+
+      <LibPackageExcludes Include="%2A%2A\%2A.dbg"/>
+      <LibPackageExcludes Include="%2A%2A\%2A.dSYM"/>
+      <LibPackageExcludes Include="%2A%2A\%2A.debug"/>
```
Comment (am11) [inline src/installer/pkg/projects/Microsoft.DotNet.ILCompiler/Microsoft.DotNet.ILCompiler.pkgproj:39]:
```suggestion
      <LibPackageExcludes Include="%2A%2A\%2A.dbg"/>
      <LibPackageExcludes Include="%2A%2A\%2A.debug"/>
      <LibPackageExcludes Include="%2A%2A\%2A.dSYM"/>
      <LibPackageExcludes Include="%2A%2A\%2A.dwarf"/>
```

**Review by jkotas — APPROVED** [review]
Thank you!

**Pull request merged**